### DISCUSSION TITLE
[ospp] Can we provide a command line terminal (CLI)？

### DISCUSSION THREAD

**Discussion by KeeProMise** [apache/incubator-hugegraph#2192] [Ideas]
Can we provide a command line terminal (CLI)？similar to redis and mysql clients, through which users can easily view database information and read and write databases

**Comment by imbajin** [comment]
Good idea, 2 points to discuss:
1. execute `gremlin/cypher` to query/add/delete data could be done in the [hubble](https://hugegraph.apache.org/docs/quickstart/hugegraph-hubble/), and it supports view `graph schema` now & **will** support view DB info and cluster info **soon**.  (**Option 1** is to expand the visualization **capability** in hubble)
2. `Tinkerpop` already support query graph data in `gremlin-console`, and HG also supports this mode, however it does not support overall graph information or specific API query for some algorithms. (We could consider extending based on it, it's **Option 2**)

Wonder to know in what scenario would normal users prefer to use a separate command line for queries?

**Comment by KeeProMise** [comment]
Thanks for your pointers, I am a beginner in graph databases, https://hugegraph.apache.org/cn/docs/quickstart/hugegraph-hubble/According to this document, I think the function is indeed much better than the command line CLI, but This document did not find guidance on how to install and start hubble.

**Reply by imbajin** [reply]
refer the README now ：https://github.com/apache/incubator-hugegraph-toolchain/blob/master/hugegraph-hubble/README.md  （docker way is the easiest way）

We do lack some doc in the website page，and prepare to fulfill it soon（it was merged into the toolchain package，u could also download the toolchain.tar.gz to get it）

And welcome to improve the hubble doc when u take a try with it~

**Reply by KeeProMise** [reply]
Okay, thank you very much. I think the two options you mentioned above are good, and there are more clients that can be supported. I think it only supports java clients at present, and it may support clients in mainstream languages such as python, go, and c++. end, so that users have more choices, and can also expand more users who are familiar with other languages.

**Reply by imbajin** [reply]
> Okay, thank you very much. I think the two options you mentioned above are good, and there are more clients that can be supported. I think it only supports java clients at present, and it may support clients in mainstream languages such as python, go, and c++. end, so that users have more choices, and can also expand more users who are familiar with other languages.

yep, also add some context for the `multi-language-client` support:
1. If the user use `gremlin` as their major graph query way, `Tinkerpop Gremlin` (our graph query language) provided a string of official [gremlin-client ](https://tinkerpop.apache.org/community.html) (python/go/js/scala/groovy...)to use
2. HugeGraph also provides some **directly**-called [RESTful API](https://hugegraph.apache.org/docs/clients/restful-api/) for some useful OLTP & OLAP algorithms, however `gremlin-client` can't call them directly, now they have to use `HTTP/Websocket` way to attach them
3. Although **Way1** & **Way2** could provide a possible(easy) way to use HG, the best way as u said we should better provide the relative **client/SDK** for users(for the perf & compatibility), but we lack enough devs to maintain them 😢 (so we need & welcome the [hg-python-client](https://github.com/apache/incubator-hugegraph/issues/79#issuecomment-1465396843))

**Comment by imbajin** [comment]
Update from gdotv's invitation:
![image](https://github.com/apache/incubator-hugegraph/assets/17706099/1037d8a5-d49e-4735-a7f7-203615bb1ceb)

@KeeProMise could take a look about it

refer https://gdotv.com/#, consider support it in our hubble(dashboard) UI? @JackyYangPassion @wanganjuan
